kuma/users/stripe_utils.py

```diff
@@ -29,47 +27,24 @@ def create_stripe_customer_and_subscription_for_user(user, email, stripe_token):
         user.stripe_customer_id = customer.id
         user.save()
 
-    if retrieve_stripe_subscription(customer) is None:
-        stripe.Subscription.create(
+    subscription = retrieve_stripe_subscription(customer)
+    if not subscription:
+        subscription = stripe.Subscription.create(
             customer=customer.id, items=[{"plan": settings.STRIPE_PLAN_ID}],
         )
 
+    return subscription
 
-def retrieve_stripe_subscription_info(user):
-    stripe_customer = (
-        stripe.Customer.retrieve(user.stripe_customer_id, expand=["default_source"],)
-        if settings.STRIPE_PLAN_ID and user.stripe_customer_id
-        else None
-    )
 
-    stripe_subscription = (
-        retrieve_stripe_subscription(stripe_customer)
-        if stripe_customer and stripe_customer.email == user.email
-        else None
-    )
-    if stripe_subscription:
-        source = stripe_customer.default_source
-        if source.object == "card":
-            card = source
-        elif source.object == "source":
-            card = source.card
-        else:
-            raise ValueError(
-                f"unexpected stripe customer default_source of type {source.object!r}"
-            )
-
-        return {
-            "next_payment_at": datetime.fromtimestamp(
-                stripe_subscription.current_period_end
-            ),
-            "brand": card.brand,
-            "expires_at": f"{card.exp_month}/{card.exp_year}",
-            "last4": card.last4,
-            # Cards that are part of a "source" don't have a zip
-            "zip": card.get("address_zip", None),
-        }
```

Comment:
The explanation for this is that I first of all needed to break it up. I need to retrieve the subscription so I can take its `.id` and stick it into `UserSubscription`. 
Technically, it might be possible to keep this function but I just thought it'd be better to A) break it up and B) this stuff of making a little dict to is exclusive to the view function. It's not a utility function. It belongs to the view only. 
I hope you can muster my refactoring.